### ISSUE TITLE
`:multi` transformer doesn't work as expected

### ISSUE THREAD

**Issue by rynkowsg** [metosin/malli#494]
**Problem**
I wrote a few tests validating my assumptions about Malli transformers and I noticed one inconsistency in values transformation using `mt/default-value-transformer)`. Please have a look at the below examples. First test with `:multi` fails, the second with `:orn` doesn't.

It is either my misunderstanding of `:multi` or transformers, or the `:multi` transformer does not visit schemas hold by `:multi`.

**Tests**

```clojure
(deftest test-default-value-transformer
  (testing "coercing with :default - in multi schema"
    (is (= {:category :book :title "Fountainhead"}
           (m/decode
            [:multi
             {:dispatch :category}
             [:book
              [:map
               [:category [:= :book]]
               [:title {:default "Fountainhead"} string?]]]
             [:video
              [:map
               [:category [:= :video]]
               [:name string?]]]]
            {:category :book}
            (mt/default-value-transformer)))))
  (testing "coercing with :default - in orn schema"
    (is (= {:category :book :title "Fountainhead"}
           (m/decode
            [:orn
             [:book
              [:map
               [:category [:= :book]]
               [:title {:default "Fountainhead"} string?]]]
             [:video
              [:map
               [:category [:= :video]]
               [:name string?]]]]
            {:category :book}
            (mt/default-value-transformer))))))
```
```
FAIL in utils.malli-test/test-default-value-transformer (malli_test.clj:147)
coercing with :default - in multi schema
Expected:
  {:category :book, :title "Fountainhead"}
Actual:
  {:category :book, -:title "Fountainhead"}
```

**Comment by rynkowsg** [comment]
I checked also checked inline decode hooks (`:decode/...` schema properties), and the problem is the same:
```clojure
(deftest decode-hooks
  (testing "in map schema"
    (is (= {:title "FOUNTAINHEAD"}
           (m/decode
            [:map
             [:title [string? {:decode/string-upper '(fn [s] (clojure.string/upper-case s))}]]]
            {:title "Fountainhead"}
            (mt/transformer {:name :string-upper})))))

  (testing "in orn schema"
    (is (= {:category :book :title "FOUNTAINHEAD"}
           (m/decode
            [:orn
             [:book
              [:map
               [:category [:= :book]]
               [:title {:decode/string-upper '(fn [s] (clojure.string/upper-case s))} string?]]]
             [:video
              [:map
               [:category [:= :video]]
               [:name string?]]]]
            {:category :book :title "Fountainhead"}
            (mt/transformer {:name :string-upper})))))

  (testing "in multi schema (one option)"
    (is (= {:category :book :title "FOUNTAINHEAD"}
           (m/decode
            [:multi
             {:dispatch :category}
             [:book
              [:map
               [:category [:= :book]]
               [:title {:decode/string-upper '(fn [s] (clojure.string/upper-case s))} string?]]]]
            {:category :book :title "Fountainhead"}
            (mt/transformer {:name :string-upper})))))

  (testing "in multi schema (two options)"
    (is (= {:category :book :title "FOUNTAINHEAD"}
           (m/decode
            [:multi
             {:dispatch :category}
             [:book
              [:map
               [:category [:= :book]]
               [:title {:decode/string-upper '(fn [s] (clojure.string/upper-case s))} string?]]]
             [:video
              [:map
               [:category [:= :video]]
               [:name string?]]]]
            {:category :book :title "Fountainhead"}
            (mt/transformer {:name :string-upper}))))))
```
```
  decode-hooks
    in orn schema
    in orn schema
    in multi schema (one option)
    in multi schema (two options) FAIL
....
FAIL in utils.malli-test/decode-hooks (malli_test.clj:252)
inline decode hooks in multi schema (two options)
Expected:
  {:category :book, :title "FOUNTAINHEAD"}
Actual:
  {:category :book, :title -"FOUNTAINHEAD" +"Fountainhead"}
```

It is worth to notice, that decoding works fine for `:multi` with only one option, but it stops working when `:multi` holds two options.

**Comment by ikitommi** [comment]
Thanks for the detailed report! There was a missing else branch on `:multi` transformers. Added your test and a fix in #495 . Deployed also to clojars:

```
➜  malli git:(master) clj -Sforce -Sdeps '{:deps {metosin/malli {:mvn/version "0.6.0-SNAPSHOT"}}}'
Downloading: metosin/malli/0.6.0-SNAPSHOT/malli-0.6.0-20210730.081527-5.pom from clojars
```

**Comment by rynkowsg** [comment]
Thank you. It works correctly now.
